### DISCUSSION TITLE
Join channels based on your main tab layout then alphabetically.

### DISCUSSION THREAD

**Discussion by RynTurtle** [Chatterino/chatterino2#4683] [Ideas] [answered]
I believe it would be nice to join your main tab layout first then alphabetically. https://i.imgur.com/e7vFbhM.png for example joins A (austinshow) almost instantly  but then T (turtoise) last with a gap of 40 seconds between them. 

I believe that having it prioritise what you have open would help  it seem like its joined quicker on reconnect notice because most people just have a main layout which they use to chat most of the time and sue the other tabs for occasional chatting.

**Comment by Felanbird** [comment]
This is only really a power user problem, and it doesn't have a power user fix, as I'd say a large % of the Chatterino userbase has never even interacted with the need to connect to multiple pools of channels.

We don't have any way of indicating what is your `main` tab, besides looking at what tab has the most channels, and in my personal usecase that would load bunch of channels I don't really care about, long before a few tabs that do load first alphabetically.

I could see it changed to loading the channels in order as they are listed, but that's really about it.

**Reply by Mm2PL** [reply]
I say that we should assume that the first tab is the 'main' tab. Or even join channels in tab order, left-to-right. Another option to look at would be saving the tab that was open when Chatterino was last used, this way the most recent tab would be joined first. If we're looking for something even more complex, which I don't think we are, we can consider bumping visible channels on the list of channels to join. I don't think hours of programming time and really high code complexity is worth 30 seconds more rate limit time each reconnect.

**Comment by 8thony** [comment, marked as answer]
- Duplicate of #5194
- Implemented since #5850
